app/models/Email.php

```diff
@@ -35,18 +35,15 @@ public static function sendEmail($key, $recipient, $to, $extra_tokens = [])
 
         $tokens = array_merge($tokens, $extra_tokens);
 
-        $subject = $email->subject;
-        $body = $email->body;
-
-        // Replace all the tokens
-        $body = str_replace(array_keys($tokens), array_values($tokens), $body);
-        $subject = str_replace(array_keys($tokens), array_values($tokens), $subject);
+        // If this is a test email, grab the content from the form. Otherwise, grab email content from the database and replace all the tokens.
+        $body = $tokens['body'] ? $tokens['body'] : str_replace(array_keys($tokens), array_values($tokens), $email->body);
+        $subject = $tokens['subject'] ? $tokens['subject'] : str_replace(array_keys($tokens), array_values($tokens), $email->subject);
```

Comment:
I think we do want these stored on the Email model itself because the tokens are for replacing things like `:first_name:` or `:last_name:` and this isn't the use case for those

Reply:
@katiecrane ah ok! The only thing I was thinking is that we don't necessarily have to store all this test data in the database since they're just test emails. Instead, we could just have one `test` model in the `emails` table with a `null` `subject` and `body` and then just take this from the form. Otherwise, it would create a new record in the the `emails` table each time a new test email is sent. But, maybe it is actually better to just save each test email in the database so that the code is always the same and get rid of this conditional?

Reply:
@chloealee if we use `firstOrCreate` when creating the test email, we will only create it if it doesn't exist, so it'll be created the first time a test email is sent and then not created on subsequent sends. Then we just keep updating the `body` and `subject` of that same database entry!

Reply:
oohhh you're totally right, thanks @katiecrane ! I wasn't seeing it being saved because I forgot to do `$email->save();` so kinda got derailed but just updated! :) Should be good for final review now!

Reply:
@chloealee oooo right! I totally didn't include that in my snippet either, good catch!